lib/psych/scalar_scanner.rb

```diff
@@ -14,10 +14,10 @@ class ScalarScanner
               |\.(nan|NaN|NAN)(?# not a number))$/x
 
     # Taken from http://yaml.org/type/int.html
-    INTEGER = /^(?:[-+]?0b[0-1_,]+          (?# base 2)
-                  |[-+]?0[0-7_,]+           (?# base 8)
-                  |[-+]?(?:0|[1-9][0-9_,]*) (?# base 10)
-                  |[-+]?0x[0-9a-fA-F_,]+    (?# base 16))$/x
+    INTEGER = /^(?:[-+]?0b[0-1_,]+                        (?# base 2)
+                  |[-+]?0[0-7_,]+                         (?# base 8)
+                  |[-+]?(?:0|[1-9]([0-9]|,[0-9]|_[0-9])*) (?# base 10)
```

Comment:
Perhaps this new grouping should also be non-capturing?

```suggestion
                  |[-+]?(?:0|[1-9](?:[0-9]|,[0-9]|_[0-9])*) (?# base 10)
```

Interactive example at: https://rubular.com/r/x8Xufg5TPMDU3X

Reply:
@olleolleolle can you check spec?